core/trino-main/src/main/java/io/trino/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -123,7 +119,7 @@ public double getUtilization()
     @Override
     public boolean isOverutilized()
     {
-        return (memoryManager.getUtilization() >= 0.5) || !state.get().canAddPages();
+        return (memoryManager.getUtilization() >= 0.5) || !stateMachine.getState().canAddPages();
```

Comment:
nit: you can add shortcut `stateMachine#canAddPages` here and in other places

Reply:
The `canAddPage` property is a property of a state and not of a state machine. `stateMachine.canAddPage` can be read as "can I add a page to state machine" vs is the current state allows new pages to be added.